src/services/expensesService.js

```diff
@@ -0,0 +1,56 @@
+let expenses = [];
+let nextId = 1;
+
+const resetExpenses = () => {
+  expenses = [];
+  nextId = 1;
+};
+
+const getAllExpenses = (userId, categories, from, to) => {
+  return expenses
+    .filter((expense) => !userId || expense.userId === +userId)
+    .filter((expense) => !categories || expense.category === categories)
```

Comment:
The filter method creates a new array, so chaining multiple filters like this can lead to unnecessary performance overhead. Consider combining these conditions into a single filter function.